rdflib/plugins/sparql/parser.py

```diff
@@ -1508,25 +1511,27 @@ def expandCollection(terms):
 UpdateUnit.ignore("#" + restOfLine)
 
 
-expandUnicodeEscapes_re = re.compile(r"\\u([0-9a-f]{4}(?:[0-9a-f]{4})?)", flags=re.I)
+expandUnicodeEscapes_re: re.Pattern = re.compile(
+    r"\\u([0-9a-f]{4}(?:[0-9a-f]{4})?)", flags=re.I
+)
 
 
-def expandUnicodeEscapes(q):
+def expandUnicodeEscapes(q: str) -> str:
     r"""
     The syntax of the SPARQL Query Language is expressed over code points in Unicode [UNICODE]. The encoding is always UTF-8 [RFC3629].
     Unicode code points may also be expressed using an \ uXXXX (U+0 to U+FFFF) or \ UXXXXXXXX syntax (for U+10000 onwards) where X is a hexadecimal digit [0-9A-F]
     """
 
-    def expand(m):
+    def expand(m: re.Match) -> str:
         try:
             return chr(int(m.group(1), 16))
-        except:  # noqa: E722
-            raise Exception("Invalid unicode code point: " + m)
+        except (ValueError, OverflowError) as e:
```

Comment:
I'm not sure if `OverflowError` could be raised from here, not that it is that much of a problem if it can't but would like to know just for interest’s sake.

Reply:
In my local testing I found that `OverflowError` could be raised for large 8-digit hexadecimal codepoint numbers - apparently because they can be larger than a 32-bit int.  I'm not sure why the max isn't 64-bit, but that might be a limitation caused by the use of `int()`.

Valid codepoints can't be larger than `\U10000000`, so it's kind of a moot point, except that due to the way the regex is written, the function will try to handle `\U99999999` and will trigger `OverflowError`.